src/workflow/ArcanistDiffWorkflow.php

```diff
@@ -1739,6 +1757,35 @@ private function getCommitMessageFromUser() {
   }
 
 
+  private function modifiesPath($paths_to_check, $changes) {
+    foreach ($changes as $changed_file => $contents) {
+      if (preg_match($paths_to_check, $changed_file)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * get revert plan by prompting developer
+   */
+  private function getRevertPlan() {
+    $ff_prompt = 'Is this diff guarded by a feature flag?';
+    $flagged = phutil_console_confirm($ff_prompt, $default_no = false);
+    if ($flagged) {
+      $flagnames = phutil_console_prompt('Flag name:');
+      return 'Guarded by feature flag ' .  $flagnames;
+    } else {
+      $reasons = $this->getConfigurationManager()->getConfigFromAnySource('differential.revert_plan_ommitted_reasons');
```

Comment:
I "externalized" the reasons for not flagging to the configuration so that we could tweak them easily without patching arcanist.  Does it make sense to do the same for any of the other strings in this fuction?  @raviagarwal7 @kageiit

Reply:
having feature flag makes sense to me. The other use case I can think of is having flipr flags which might still fall in the feature flag criteria. might be a good idea to have the current as default that override with the one present in the config if any.

Reply:
Ok, in an attempt to not over-engineer this, I'll make it more configurable in a follow-up PR